proton.py

```diff
@@ -0,0 +1,509 @@
+#script to launch Wine with the correct environment
+
+from __future__ import print_function
+
+import filecmp
+import json
+import os
+import shutil
+import struct
+import subprocess
+import sys
+import tarfile
+
+from filelock import FileLock, Timeout
+
+#To enable debug logging, copy "user_settings.sample.py" to "user_settings.py"
+#and edit it if needed.
+
+CURRENT_PREFIX_VERSION="3.7-2"
+
+PFX="Proton: "
+
+if "Darwin" in os.uname()[0]:
+    ld_path_var = "DYLD_LIBRARY_PATH"
+else:
+    ld_path_var = "LD_LIBRARY_PATH"
+
+def log(msg):
+    sys.stderr.write(PFX + msg + os.linesep)
+    sys.stderr.flush()
+
+def upgrade_pfx(old_ver):
+    if old_ver == CURRENT_PREFIX_VERSION:
+        return
+
+    log("Upgrading prefix from " + str(old_ver) + " to " + CURRENT_PREFIX_VERSION + " (" + os.environ["STEAM_COMPAT_DATA_PATH"] + ")")
+
+    if old_ver == None:
+        return
+
+    old_proton_ver, old_prefix_ver = old_ver.split('-')
+
+    if old_proton_ver == "3.7" and old_prefix_ver == "1":
+        if not os.path.exists(prefix + "/drive_c/windows/syswow64/kernel32.dll"):
+            #shipped a busted 64-bit-only installation on 20180822. detect and wipe clean
+            log("Detected broken 64-bit-only installation, re-creating prefix.")
+            shutil.rmtree(prefix)
+
+def run_wine(args):
+    subprocess.call(args, env=env, stderr=lfile)
+
+def makedirs(path):
+    try:
+        os.makedirs(path)
+    except:
+        #already exists
+        pass
+
+def real_copy(src, dst):
+    if os.path.islink(src):
+        os.symlink(os.readlink(src), dst)
+    else:
+        shutil.copy(src,dst)
+
+def mergedirs(src, dst):
+    for src_dir, dirs, files in os.walk(src):
+        dst_dir = src_dir.replace(src, dst, 1)
+        if not os.path.exists(dst_dir):
+            os.makedirs(dst_dir)
+        for dir_ in dirs:
+            src_file = os.path.join(src_dir, dir_)
+            dst_file = os.path.join(dst_dir, dir_)
+            if os.path.islink(src_file) and not os.path.exists(dst_file):
+                real_copy(src_file, dst_file)
+        for file_ in files:
+            src_file = os.path.join(src_dir, file_)
+            dst_file = os.path.join(dst_dir, file_)
+            if not os.path.exists(dst_file):
+                real_copy(src_file, dst_file)
+
+if not ("STEAM_COMPAT_DATA_PATH" in os.environ):
+    log("No compat data path?")
+    sys.exit(1)
+
+basedir = os.path.dirname(os.path.realpath(__file__))
```

Comment:
This is the only bit of code that I modified from the main python file, this will get `proton.py`'s directory which is what `sys.argv[0]` did previously, except this way `proton.py` can be called using `python proton.py` which wasn't possible before because the python binary is located elsewhere.